dataService.js

```diff
@@ -0,0 +1,70 @@
+let combinedProductMap = null
+
+class DataService {
```

Comment:
Enclosed most of the code in this class

Reply:
I prefer to don't go for Class types unless you are really comfortable with that.